lib/cdo/pegasus/graphics.rb

```diff
@@ -1,6 +1,12 @@
 require 'rmagick'
 
-def load_manipulated_image(path,mode,width,height=nil)
+MAX_DIMENSION = 2880
+
+def load_manipulated_image(path, mode, width, height = nil)
+
+  width = [MAX_DIMENSION, width].min
```

Comment:
Yeah, I didn't bother with finding the correct aspect ratio.  If you think I should add that I can.